app/models/proposal_keyword_recommendation.rb

```diff
@@ -0,0 +1,22 @@
+class ProposalKeywordRecommendation < ApplicationRecord
+  # this class/table is for tracking recommendations for Collection Draft
+  # Proposals ONLY
+  belongs_to :collection_draft_proposal, class_name: 'CollectionDraftProposal', foreign_key: 'draft_proposal_id', inverse_of: :keyword_recommendations, required: true
+
+  before_create :proposal_mode_enabled?, :gkr_enabled?
+  before_save :proposal_mode_enabled?, :gkr_enabled?
+  # Need to bypass this before action to delete when publishing while in dev
+  before_destroy :proposal_mode_enabled? unless Rails.env.development? || Rails.env.test?
+
+  private
+
+  def proposal_mode_enabled?
+    throw(:abort) unless Rails.configuration.proposal_mode
```

Comment:
another way to write this could be 
```
def proposal_mode_enabled?
     Rails.configuration.proposal_mode || throw(:abort)
end
```
But I am not entirely sure. If `throw(:abort)` ends the method execution (and `true` is not returned) then my suggestion would work. If the `throw(:abort)...` line is executed and the method still returns the last line = `true` (independent of the result of the first line) then my suggestion would not work. Either way, up to you